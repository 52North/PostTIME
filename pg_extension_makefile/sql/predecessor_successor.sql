SELECT predecessor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2011/2014'),
'POLYGON((3 3,0 10,10 10,10 3,3 3))',
posttime_in('2015/2016'));

SELECT predecessor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2013/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2010/2012'));

SELECT predecessor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2012/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2015/2016'));

SELECT successor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2011/2014'),
'POLYGON((3 3,0 10,10 10,10 3,3 3))',
posttime_in('2008/2010'));

SELECT successor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2013/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2010/2012'));

SELECT successor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2012/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2015/2016'));