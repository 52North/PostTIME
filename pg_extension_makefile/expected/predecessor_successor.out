SELECT predecessor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2011/2014'),
'POLYGON((3 3,0 10,10 10,10 3,3 3))',
posttime_in('2015/2016'));
 predecessor 
-------------
 f
(1 row)

SELECT predecessor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2013/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2010/2012'));
 predecessor 
-------------
 t
(1 row)

SELECT predecessor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2012/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2015/2016'));
 predecessor 
-------------
 f
(1 row)

SELECT successor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2011/2014'),
'POLYGON((3 3,0 10,10 10,10 3,3 3))',
posttime_in('2008/2010'));
 successor 
-----------
 f
(1 row)

SELECT successor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2013/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2010/2012'));
 successor 
-----------
 f
(1 row)

SELECT successor( 
'POLYGON((1 1,1 20,20 20,20 1,1 1))',
posttime_in('2012/2014'),
'POLYGON((1 1,5 10,20 20,20 1,1 1))',
posttime_in('2015/2016'));
 successor 
-----------
 t
(1 row)

